docs/develop-an-agent.md

```diff
@@ -0,0 +1,423 @@
+
+1. Install the Microsoft GPG public key
+
+    ```bash
+    curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg
+    sudo cp ./microsoft.gpg /etc/apt/trusted.gpg.d/
+    ```
+
+2. On Ubuntu 18.04
+
+    ```bash
+    curl https://packages.microsoft.com/config/ubuntu/18.04/multiarch/prod.list > ./microsoft-prod.list
+    sudo cp ./microsoft-prod.list /etc/apt/sources.list.d/
+    ```
+
+3. On Raspberry Pi OS Stretch
+
+    ```bash
+    curl https://packages.microsoft.com/config/debian/stretch/multiarch/prod.list > ./microsoft-prod.list
+    sudo cp ./microsoft-prod.list /etc/apt/sources.list.d/
+    ```
+
+#### Install for an alternative distro / architecture
+
+A pre-built package for alternative distributions / architectures is available with each release of IoT Edge (v1.2 and later). Download and install the `aziot-identity-service` package for your respective distro / architecture from [IoT Edge v1.2 release collateral](https://github.com/Azure/azure-iotedge/releases/tag/1.2.0).
+
+Using Ubuntu 20.04 amd64 as an example:
+
+```bash
+wget https://github.com/Azure/azure-iotedge/releases/download/1.2.0/aziot-identity-service_1.2.0-1_ubuntu20.04_amd64.deb -o aziot-identity-service.deb
+
+sudo apt install ./aziot-identity-service.deb
+```
+
+### Configuration for Provisioning
+
+For demonstration purposes we <u>m</u>anually <u>p</u>rovision the device with a connection string, but the identity service can be configured to using any of the following methods.
+1. manually using a connection string
+1. manually using X.509 certificates
+1. via DPS using X.509 certificates (self-signed or CA-signed)
+1. via DPS using TPS attestation
+
+We'll use the `config mp` convenience command:
+
+* If you have installed _IoT Edge v1.2+_ then use the `iotedge` CLI.
+
+    ```bash
+    sudo iotedge config mp -c "<conn_string>"
+
+    sudo iotedge config apply
+    ```
+
+* Otherwise, if only the _Identity Service_ is installed use the more basic `aziotctl` CLI.
+
+    ```bash
+    sudo aziotctl config mp -c "<conn_string>"
+
+    sudo aziotctl config apply
+    ```
+
+    > **Note**
+    >
+    > The `aziotctl` CLI isn't aware of IoT Edge and only provides a subset of functionality found in the `iotedge` CLI. Use the `iotedge` CLI when available.
+
+#### Alternative Provisioning Methods
+
+For the other provisioning methods, start by creating a config file. Modify it to set appropriate values and then apply the new configuration when you're done.
+
+For example:
+
+* If you have installed _IoT Edge v1.2+_ then use the `/etc/aziot/config.toml.edge.template`.
+
+    ```bash
+    sudo cp /etc/aziot/config.toml.edge.template /etc/aziot/config.toml
+
+    # make configuration edits
+    nano /etc/aziot/config.toml
+
+    sudo iotedge config apply
+    ```
+
+* Otherwise, if only the _Identity Service_ is installed use the more basic `/etc/aziot/config.toml.template` that does not include any IoT Edge-specific configuration options.
```

Comment:
```suggestion
* Otherwise, if only the _Identity Service_ is installed, use the more basic `/etc/aziot/config.toml.template` that does not include any IoT Edge-specific configuration options.
```